test/integration/confirmations/signatures/permit.test.tsx

```diff
@@ -0,0 +1,239 @@
+  ...jest.requireActual('../../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+}));
+
+const mockedBackgroundConnection = jest.mocked(backgroundConnection);
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+const getMetaMaskStateWithUnapprovedPersonalSign = (accountAddress: string) => {
+  const pendingPermitId = 'eae47d40-42a3-11ef-9253-b105fa7dfc9c';
+  const pendingPermitTime = new Date().getTime();
+  return {
+    ...mockMetaMaskState,
+    preferences: {
+      ...mockMetaMaskState.preferences,
+      redesignedConfirmationsEnabled: true,
+    },
+    unapprovedTypedMessages: {
+      [pendingPermitId]: {
+        id: pendingPermitId,
+        status: 'unapproved',
+        time: pendingPermitTime,
+        type: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA,
+        securityProviderResponse: null,
+        msgParams: {
+          from: accountAddress,
+          version: 'v4',
+          data: `{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Permit":[{"name":"owner","type":"address"},{"name":"spender","type":"address"},{"name":"value","type":"uint256"},{"name":"nonce","type":"uint256"},{"name":"deadline","type":"uint256"}]},"primaryType":"Permit","domain":{"name":"MyToken","version":"1","verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC","chainId":1},"message":{"owner":"${accountAddress}","spender":"0x5B38Da6a701c568545dCfcB03FcB875f56beddC4","value":3000,"nonce":0,"deadline":50000000000}}`,
+          origin: 'https://metamask.github.io',
+          siwe: { isSIWEMessage: false, parsedMessage: null },
+          signatureMethod: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA_V4,
+        },
+      },
+    },
+    unapprovedTypedMessagesCount: 1,
+    pendingApprovals: {
+      [pendingPermitId]: {
+        id: pendingPermitId,
+        origin: 'origin',
+        time: pendingPermitTime,
+        type: ApprovalType.EthSignTypedData,
+        requestData: {
+          from: accountAddress,
+          version: 'v4',
+          data: `{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Permit":[{"name":"owner","type":"address"},{"name":"spender","type":"address"},{"name":"value","type":"uint256"},{"name":"nonce","type":"uint256"},{"name":"deadline","type":"uint256"}]},"primaryType":"Permit","domain":{"name":"MyToken","version":"1","verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC","chainId":1},"message":{"owner":"${accountAddress}","spender":"0x5B38Da6a701c568545dCfcB03FcB875f56beddC4","value":3000,"nonce":0,"deadline":50000000000}}`,
+          origin: 'https://metamask.github.io',
+          signatureMethod: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA_V4,
+          metamaskId: pendingPermitId,
+        },
```

Comment:
requestData seems to be exactly the same as `msgParams` above... if that is the case, can we simplify this and avoid duplication?

Reply:
done